### PR TITLE
fix update data in redux when editing a post

### PR DESCRIPTION
**Pull request by yuriimartynenko**

### CODE REVIEW

File: client/src/store/post/actions.ts [hunk 1]
```diff
@@ -13,6 +14,7 @@ import {
 } from './actionTypes';
 import { IPost } from './reducers';
 import { decrementPostCount } from '../profile/actions';
+import { statement } from '@babel/template';
```
Comment (InnaKutsik) [inline client/src/store/post/actions.ts]: ????

File: client/src/store/post/actions.ts [hunk 2]
```diff
@@ -50,9 +52,10 @@ export const showPost = (post: any): { type: string, payload: any } => ({
     payload: post,
 });
 
-export const editDescriptionForPost = (description: any): { type: string, payload: any } => ({
+export const editDescriptionForPost = (description: any): { type: string, payload: any, postId: string } => ({
     type: EDIT_DESCRIPTION_FOR_POST,
     payload: description,
+    postId: store.getState().userPosts.selectedPost._id,
```
Comment (InnaKutsik) [inline client/src/store/post/actions.ts]: або передавай агрументом, або роби через redux-thunk (там другий аргумент після dispatch, це getState)

File: client/src/store/post/reducers.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ const defaultState = {
 
 export const postReducer = (
     state: any = defaultState,
-    action: { type: string, payload: any, loading: boolean }): any => {
+    action: { type: string, payload: any, loading: boolean, postId: string }): any => {
```
Comment (InnaKutsik) [inline client/src/store/post/reducers.ts]: reducer is a simple function which has 2 arguments - type: string, payload

remove loading: boolean, postId: string

File: client/src/store/post/actions.ts (hunk 1 above)
Comment (yuriimartynenko) [reply client/src/store/post/actions.ts]: I don't even know where it's come from))
fixed

File: client/src/store/post/reducers.ts (hunk 3 above)
Comment (yuriimartynenko) [reply client/src/store/post/reducers.ts]: Okay. Fixed

File: client/src/store/post/actions.ts (hunk 2 above)
Comment (yuriimartynenko) [reply client/src/store/post/actions.ts]: Done

File: client/src/store/post/reducers.ts [hunk 4]
```diff
@@ -92,10 +92,12 @@ export const postReducer = (
         case EDIT_DESCRIPTION_FOR_POST:
             return {
                 ...state,
-                selectedPost: {
-                    ...state.selectedPost,
-                    description: action.payload,
-                },
+                posts: state.posts.map((post: any) => {
+                    if (post._id === action.payload.postId) {
+                        post.description = action.payload.description;
```
Comment (InnaKutsik) [inline client/src/store/post/reducers.ts]: return { ...post, description: action.payload.description }

Comment (yuriimartynenko) [reply client/src/store/post/reducers.ts]: fixed

**Pull request merged**